seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/checkpoint/operation/CheckpointFinishedOperation.java

```diff
@@ -50,21 +51,23 @@ public int getClassId() {
 
     @Override
     protected void writeInternal(ObjectDataOutput out) throws IOException {
-        out.writeLong(jobId);
-        out.writeLong(pipelineId);
         out.writeLong(checkpointId);
+        out.writeObject(taskInfo);
+        out.writeBoolean(successful);
     }
 
     @Override
     protected void readInternal(ObjectDataInput in) throws IOException {
-        jobId = in.readLong();
-        pipelineId = in.readLong();
         checkpointId = in.readLong();
+        taskInfo = in.readObject(TaskInfo.class);
+        successful = in.readBoolean();
     }
 
     @Override
     protected PassiveCompletableFuture<?> doRun() throws Exception {
-        // TODO: Notifies all tasks of the pipeline about the status of the checkpoint
+        SeaTunnelServer server = getService();
+        Task task = server.getTaskExecutionService().getExecutionContext(taskInfo.getTaskGroupId())
```

Comment:
Please note that this may cause a `NullPointException` because `server.getTaskExecutionService().getExecutionContext(taskInfo.getTaskGroupId())` may return `null`. So I suggest you add retry at the place where this operation is called.

Reply:
thx, done.